executor.go

```diff
@@ -275,6 +294,114 @@ func (e *Executor) provideEventLogger(t *Task) *Logger {
 	}
 }
 
+func (e *Executor) parseTaskOptionsListToMap(taskName string, flags []string) map[string]FlagArg {
+	taskFlagsMap := make(map[string]FlagArg)
+
+	for _, flag := range flags {
+		var key string
+		var val string
+		splitFlag := strings.Split(flag, "=")
+
+		if len(splitFlag) == 2 {
+			// If the passed flag has an "=" in it, assume that it is a variable flag
+			// (e.g. --var="val")
+			key = splitFlag[0]
+			val = splitFlag[1]
+		} else if len(splitFlag) == 1 {
+			// If the passed flag does not have an "=" in it, assume that it is a bool flag
+			// (e.g. --true)
+			key = splitFlag[0]
+		} else {
+			// If the passed flag has >1 "=" in it, this is invalid syntax
+			panic(fmt.Sprintf("Invalid flag syntax for %s: `%s`", taskName, flag))
+		}
+
+		key, err := e.getVerifiedFlagKey(taskName, splitFlag[0])
+		if err != nil {
+			panic(fmt.Sprintf("Unsupported flag passed to %s: `%s`. See error: %s", taskName, key, err))
+		}
+
+		// At this point, we should have verified that this flag is supported in `getVerifiedFlagKey`
+		taskFlag := e.taskFlagsRegistry[taskName][key]
+		flagValErrMsg := fmt.Sprintf("The value for this flag is `%s`. Please use `%s`", taskFlag.ValueType, flagTypeToGetter[taskFlag.ValueType])
+		flagArg := FlagArg{
+			Value: val,
+			BoolVal: func() bool {
+				if taskFlag.ValueType != "bool" {
+					panic(flagValErrMsg)
+				}
+
+				return true
+			},
+			IntVal: func() int {
+				if taskFlag.ValueType != "int" {
+					panic(flagValErrMsg)
+				}
+
+				num, err := strconv.Atoi(val)
+				if err != nil {
+					// Handle error - val was invalid int
```

Comment:
Should we panic here? Probably yes since this means somebody passed a non-int value into a flag that should only accept int

e.g.
```
 --intKey="123a" --> panic
 --intKey="123" --> ok
```